plugins/home/src/api/LocalStorageVisitsApi.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { IdentityApi } from '@backstage/core-plugin-api';
+import { Visit } from './VisitsApi';
+import { VisitsApiFactory } from './VisitsApiFactory';
+
+/** @public */
+export type LocalStorageVisitsApiOptions = {
+  limit?: number;
+  identityApi: IdentityApi;
+};
+
+/**
+ * @public
+ * This is a reference implementation of VisitsApi using window.localStorage.
+ */
+export class LocalStorageVisitsApi extends VisitsApiFactory {
```

Comment:
Could this local storage be an implementation of the Storage API? I have the feeling that this could be a `storageApi` implementation injected into the CoreStorageApi instead.

Reply:
That is a good point. This also puts into new light your [first comment](https://github.com/backstage/backstage/pull/19643#discussion_r1328989834) since I could always rely on a storageApi implementation always.
I will investigate this further and create a followup PR.